src/test/java/uk/gov/pay/adminusers/service/SecondFactorAuthenticatorTest.java

```diff
@@ -1,22 +1,37 @@
 package uk.gov.pay.adminusers.service;
 
+import com.warrenstrange.googleauth.GoogleAuthenticatorConfig;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
+import java.time.Clock;
+
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class SecondFactorAuthenticatorTest {
 
     @Rule
     public ExpectedException thrown = ExpectedException.none();
 
     private SecondFactorAuthenticator secondFactorAuthenticator;
+    private Clock clock;
+    private GoogleAuthenticatorConfig authConfig;
+    private long baseTime;
     @Before
     public void before() throws Exception {
-        secondFactorAuthenticator = new SecondFactorAuthenticator();
+        baseTime = Clock.systemDefaultZone().millis();
+        authConfig = new GoogleAuthenticatorConfig.GoogleAuthenticatorConfigBuilder()
+                .setWindowSize(9)
+                .setTimeStepSizeInMillis(30000)
```

Comment:
I think these magic numbers should be extracted into constants:

```
    private static final Instant INITIAL_TIME = Instant.now();
    private static final Duration TIME_STEP = Duration.of(30, SECONDS);
    private static final int PAST_OR_FUTURE_WINDOWS_TO_CHECK = 4;
    private static final int PAST_PRESENT_AND_FUTURE_WINDOWS_TO_CHECK = PAST_OR_FUTURE_WINDOWS_TO_CHECK * 2 + 1;
```

These constants could then be used in the tests to remove magic numbers there.